File: Arrays/448. Find All Numbers Disappeared in an Array/448. Find All Numbers Disappeared in an Array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        ans = []
        
        for num in nums:
            i = abs(num) - 1
            nums[i] = -1 * abs(nums[i])
        
        for i, num in enumerate(nums):
            if num > 0:
                ans.append(i + 1)
        
        return ans

# Time: O(n)
# Space: O(1)

File: Arrays/448. Find All Numbers Disappeared in an Array/448. Find All Numbers Disappeared in an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
  let ans = [];

  for (let i = 0; i < nums.length; i++) {
    const idx = Math.abs(nums[i]) - 1;
    nums[idx] = -1 * Math.abs(nums[idx]);
  }

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] > 0) ans.push(i + 1);
  }

  return ans;
};

// Time: O(n)
// Space: O(1)
